crates/telemetry/src/env.rs

```diff
@@ -0,0 +1,48 @@
+/// Returns a boolean indicating if the OTEL layer should be enabled.
+///
+/// It is considered enabled if any of the following environment variables are set and not empty:
+/// - `OTEL_EXPORTER_OTLP_ENDPOINT`
+/// - `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT`
+/// - `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT`
+///
+/// Note that this is overridden if OTEL_SDK_DISABLED is set and not empty.
+pub(crate) fn otel_enabled() -> bool {
+    const ENABLING_VARS: &[&str] = &[
+        "OTEL_EXPORTER_OTLP_ENDPOINT",
+        "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT",
+        "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT",
+    ];
+    ENABLING_VARS
+        .iter()
+        .any(|key| std::env::var_os(key).is_some_and(|val| !val.is_empty()))
+}
+
+/// Returns a boolean indicating if the OTEL SDK should be disabled for all signals.
+///
+/// It is considered disabled if the environment variable `OTEL_SDK_DISABLED` is set and not empty.
+pub(crate) fn otel_sdk_disabled() -> bool {
+    std::env::var_os("OTEL_SDK_DISABLED").is_some_and(|val| !val.is_empty())
+}
+
+/// The protocol to use for OTLP exporter.
+pub(crate) enum OtlpProtocol {
+    Grpc,
+    HttpProtobuf,
+    HttpJson,
+}
+
+impl OtlpProtocol {
+    /// Returns the protocol to be used for exporting traces.
+    pub(crate) fn traces_from_env() -> Self {
```

Comment:
The `traces_from_env` name read a bit oddly to me because this seems to be getting the _protocol_ from the environment.  Is the intended contrast "the protocol to use for traces" vs a future "the protocol to use for metrics"?

Reply:
> Is the intended contrast "the protocol to use for traces" vs a future "the protocol to use for metrics"?

Yes. Very open to better names.

Reply:
`traces_protocol_from_env`? It's slightly redundant, but also more obvious.